beacon-chain/blockchain/process_block.go

```diff
@@ -158,16 +158,10 @@ func (s *Service) onBlock(ctx context.Context, signed *ethpb.SignedBeaconBlock,
 	}
 
 	// Update finalized check point.
-	if (postState.FinalizedCheckpointEpoch() > s.finalizedCheckpt.Epoch) || (featureconfig.Get().UpdateHeadTimely && newFinalized) {
-		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
```

Comment:
This chunk of code is repeated again in `updateFinalized` so we remove it over here.

Reply:
Is it ok to ignore the effect of `featureconfig.Get().UpdateHeadTimely` flag? I mean we now have a slightly different semantics: if the first block works (`newFinalized` is true), then this block will also execute, but previously we needed the flag to be on, as well.

Reply:
oh, never mind it is a disjunction! so no change in semantics.

Reply:
The head timely flag is not needed. In the event that it is not applied we still execute the code block due to the first condition. Which basically checks if finalized checkpoint has changed.